tests/phpunit/tests/html-api/wpHtmlProcessorBreadcrumbs.php

```diff
@@ -352,6 +352,45 @@ public function data_html_target_with_breadcrumbs() {
 		);
 	}
 
+	/**
+	 * @ticket {TICKET_NUMBER}
+	 *
+	 * @dataProvider data_html_with_breadcrumbs_of_various_specificity
+	 *
+	 * @param string   $html_with_target_node HTML with a node containing a "target" attribute.
+	 * @param string[] $breadcrumbs           Breadcrumbs to test at the target node.
+	 * @param bool     $should_match          Whether the target node should match the breadcrumbs.
+	 */
+	public function test_reports_if_tag_matches_breadcrumbs_of_various_specificity( $html_with_target_node, $breadcrumbs, $should_match ) {
+		$processor = WP_HTML_Processor::createFragment( $html_with_target_node );
+		while ( $processor->next_tag() && null === $processor->get_attribute( 'target' ) ) {
+			continue;
+		}
+
+		$matches = $processor->matches_breadcrumbs( $breadcrumbs );
+		$path    = implode( ', ', $breadcrumbs );
+		if ( $should_match ) {
+			$this->assertTrue( $matches, "HTML tag {$processor->get_tag()} should have matched breadcrumbs but didn't: {$path}." );
+		} else {
+			$this->assertFalse( $matches, "HTML tag {$processor->get_tag()} should not have matched breadcrumbs but did: {$path}." );
+		}
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[].
+	 */
+	public function data_html_with_breadcrumbs_of_various_specificity() {
+		return array(
+			'Inner IMG'                  => array( '<div><span><figure><img target></figure></span></div>', array( 'span', 'figure', 'img' ), true ),
+			'Inner IMG wildcard'         => array( '<div><span><figure><img target></figure></span></div>', array( 'span', '*', 'img' ), true ),
+			'Inner IMG no wildcard'      => array( '<div><span><figure><img target></figure></span></div>', array( 'span', 'img' ), false ),
+			'Full specification'         => array( '<div><span><figure><img target></figure></span></div>', array( 'html', 'body', 'div', 'span', 'figure', 'img'), true ),
+			'Invalid Full specification' => array( '<div><span><figure><img target></figure></span></div>', array( 'html', 'div', 'span', 'figure', 'img'), false ),
```

Comment:
```suggestion
			'Full specification'         => array( '<div><span><figure><img target></figure></span></div>', array( 'html', 'body', 'div', 'span', 'figure', 'img' ), true ),
			'Invalid Full specification' => array( '<div><span><figure><img target></figure></span></div>', array( 'html', 'div', 'span', 'figure', 'img' ), false ),
```

Fix PHPCS failure

Reply:
Thanks @mukeshpanchal27 - I've incorporated your change into ab6952c216

Please feel free to overlook code-styling issues for now, both because this isn't ready for that stage of review, and also because the CI job should point all these out, and I wouldn't want you to spend your time doing what the automated system will do.